### PR TITLE
Fix site re assignement

### PR DESCRIPTION
**Pull request by MaximeCulea**
Hi @johnjamesjacoby,

I managed to clean the code for this PR : https://github.com/stuttter/wp-multi-network/pull/74
It is ready for merge under 1.7.1 tag.

I have tested this on a website which was already using 74's PR version. So re-assignement, add and take-off blog to/from site, is now wonderfully working !
I hope it's all good for you, and can't wait for your feedback :smiley:

Cheers

### CODE REVIEW

**Comment by MaximeCulea** [comment]
Ho @johnjamesjacoby,
Any news? I am waiting this feature to be merged for a huge site ;)
In advance, thx!

**Comment by MaximeCulea** [comment]
Hi @johnjamesjacoby @rmccue,
Any news about this merge ? :(
Best,

File: includes/functions.php [hunk 1]
```diff
@@ -744,6 +757,12 @@ function move_site( $site_id = 0, $new_network_id = 0 ) {
 		restore_current_network();
 	}
 
+	// Update blog's siteurl and home options
+	if ( 0 !== $new_network_id ) {
+		update_blog_option( $site_id, 'siteurl', str_replace( "/{$site->domain}/", "/{$new_network_domain}/", get_blog_option( $site_id, 'siteurl' ) ) );
+		update_blog_option( $site_id, 'home', str_replace( "/{$site->domain}/", "/{$new_network_domain}/", get_blog_option( $site_id, 'home' ) ) );
+	}
+
```
Comment (johnjamesjacoby) [inline includes/functions.php]: How does this work on subdomain installations?

File: includes/functions.php [hunk 2]
```diff
@@ -38,6 +38,46 @@ function get_networks() {
...
+	function has_path_for_network( $network_id, $path ) {
+		global $wpdb;
+
+		// Check for existing network
+		$network = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM {$wpdb->blogs} WHERE site_id = %d AND path = %s LIMIT 1", $network_id, $path ) );
+
+		return ! empty( $network );
+	}
+endif;
+
```
Comment (johnjamesjacoby) [inline includes/functions.php]: How do these work on subdomain installations?

WordPress 4.6 will have query classes and functions to help with these, so we might be better off using them instead?

**Comment by johnjamesjacoby** [comment]
I'm looking at this now.

I like the check to avoid having duplicate domain & path combinations, but I'm not sure if moving sites between networks should force-update `home`, `site_url`, and/or `domain` & `path`.

I have definitely used `move_site()` just to move sites between networks, only to switch over their `site_id` and without wanting to update anything else.

**Comment by inderpreet99** [comment]
At Boston University, we definitely see the need for url/domain to be updated to fit the new network. That's how we've used the plugin in the past. To us, each domain represents a new network. Although networks can have specific paths, we have usually left them at the default `/`.

If the current setup works for some and this PR works for others, we should have a setting in the plugin to support both workflows.

@johnjamesjacoby, maybe we could have a visual checkbox on the `Move Site` screen and/or a constant similar to how RESCUE_ORPHANED_BLOGS works.

Thanks.

**Comment by johnjamesjacoby** [comment]
The idea of a checkbox to convert domains or paths is a *great* idea. We will need a more sophisticated domain & path diff engine, with some way to visually dry-run the change and make sure that's what the network admin intended to do.

If anyone wants to help, I'd love for this to be part of WP Multi Network.

**Comment by MaximeCulea** [comment]
@inderpreet99 you are totally right! The fact that it well working for ones it's mainly cause people use subdomains. For my needs, as you, I also needed sub-folder install.

Now, talking about this checkbox, I think it can be a good idea. But regardless to the network install, why not focusing on the **SUBDOMAIN_INSTALL** value ?

**Comment by johnjamesjacoby** [comment]
There's a 50% chance someone chose sub-directories, and WordPress supports any combination between sites & networks, so the assumption that subdomains and/or paths should be reassigned based on the root-site is not a safe one.

We'd need to rely on the network admin to say: yes, the site URL is this, and I need it to be this.

**Comment by MaximeCulea** [comment]
Hi @johnjamesjacoby,
So if I understand well, it has been fixed under version 1.8.1 ?
Thx

**Comment by johnjamesjacoby** [comment]
This PR has a few issues, so I'm going to close it and look into a different approach.

@inderpreet99, @MaximeCulea, thanks for the discussion and ideas. I'll be implementing something that works for all installation types. 🎱

**Pull request closed without merging**